bulldozer/merge.go

```diff
@@ -57,6 +57,39 @@ func NewGitHubMerger(client *github.Client) Merger {
 }
 
 func (m *GitHubMerger) Merge(ctx context.Context, pullCtx pull.Context, method MergeMethod, msg CommitMessage) (string, error) {
+	if method == FastForwardOnly {
+		return m.ffOnlyMerge(ctx, pullCtx)
+	}
+
+	return m.defaultMerge(ctx, pullCtx, method, msg)
+}
+
+func (m *GitHubMerger) ffOnlyMerge(ctx context.Context, pullCtx pull.Context) (string, error) {
+	// ff-only merge is accomplished by calling Git.UpdateRef with the force parameter set to false, and the new commit hash for the base branch's pointer
+
+	base, _ := pullCtx.Branches()
+
+	ref, _, err := m.client.Git.GetRef(ctx, pullCtx.Owner(), pullCtx.Repo(), fmt.Sprintf("refs/heads/%s", base))
+	if err != nil {
+		return "", err
```

Comment:
style: let's wrap these with `errors.Wrap` to attach stack information since they are coming from third-party code

Reply:
will do